src/components/GameItem.js

```diff
@@ -4,22 +4,51 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isSpotted:false,
+    };
+  }
+  
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
+  onItemClick = () => {
+    this.setState({
+      isSpotted: true,
+    });
+    if (this.props.type === 'litter') {
+      this.props.updateScoreCallback();
+    }
+    
+  }
+
   render() {
+
+    let spottedType = undefined;
+    if (this.state.isSpotted) {
+      if (this.props.type === 'litter') {
+        spottedType = 'spotted-litter';
```

Comment:
Extremely minor nitpick: Since this code is included in `render`, it could be run many times with the same result. A slightly cleaner solution would be to put this logic in the click handler, store the class name in `state`, and read that value here instead of recomputing it.